### PR TITLE
Windows builds on GHA

### PR DESCRIPTION
**Pull request by cmacdonald**
Ian,

As requested...

I'm not sure how a native Windows build got into the Github Actions file. I did have a fork that attempted this (https://github.com/cmacdonald/trec_eval/tree/windows_msvc), but that seemingly wasnt merged.

I've redone some of the work from that branch in this fork, esp using git submodules rather than copying of files. It now builds, but the test batch file doesnt run - I'll look at that again soon. I'm putting the PR so you can start to have a look at this. I'll figure out what is wrong with the test case in the near future, before merging.

Craig

### CODE REVIEW

**Comment by cmacdonald** [comment]
Ok, this now appears to build and test out ok. Over to you for review @isoboroff. Seems mostly to have been about identifying and copying over changes from that other branch.

**Comment by isoboroff** [comment]
If only C compilers had like a switch to compile against what C was in 1984.

File: get_prefs.c [hunk 1]
```diff
@@ -152,14 +152,15 @@ int te_get_prefs(EPI * epi, char *text_prefs_file, ALL_REL_INFO * all_rel_info)
     TEXT_PREFS *text_prefs_ptr;
 
     /* Read entire file into memory */
-    if (-1 == (fd = open(text_prefs_file, 0)) ||
```
Comment (isoboroff) [inline get_prefs.c:155]: The change for the open() test is not correct.  open(2) returns -1 on fail.  You might be thinking fopen().

**Review by isoboroff — APPROVED** [review]
Did not know Windows didn't have open(2).

File: get_prefs.c (hunk 1 above)
Comment (isoboroff) [reply get_prefs.c:155]: Eyeball fail... the goal of these is to change open et al calls to fopen and friends

**Comment by cmacdonald** [comment]
> Did not know Windows didn't have open(2).

I think it does, but you can't use the filehandle in exactly the same way - the lseek or something else in that expression fails.

Good we got this merged :-) We should make a Github Action that can build the windows .exe file and put it somewhere accessible to the general public.

**Pull request merged**